.github/workflows/actions.yml

```diff
@@ -0,0 +1,36 @@
+# Lint GitHub Actions for common security issues using zizmor.
+# Docs: https://woodruffw.github.io/zizmor
+
+name: Lint GitHub Actions
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install zizmor
+        run: python -m pip install zizmor
+
+      - name: List installed packages
+        run: python -m pip freeze
+
+      - name: Lint GitHub Actions
+        run: make check-actions
+        env:
+          # Set GH_TOKEN to allow zizmor to check online vulnerabilities
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Th GitHub token will not be (and should not be) available to PRs from external forks.

Reply:
According to GitHub docs on [Using secrets in a workflow](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow):

> With the exception of `GITHUB_TOKEN`, secrets are not passed to the runner when a workflow is triggered from a forked repository.

For a `pull_request` trigger, the token provides read only privileges if the runner is triggered from a forked repo ([Permissions for the GITHUB_TOKEN](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)).

So I think this is safe. Even if the PR is coming from a forked repo, the `GITHUB_TOKEN` will allow `zizmor` to check for online vulnerabilities.

This action is similar to the one in zizmor's docs: https://woodruffw.github.io/zizmor/usage/#use-in-github-actions